main.py

```diff
@@ -2,61 +2,77 @@
 from services import Services
 from application import App
 from PyQt5.QtWidgets import QApplication
-
+import logging
+from datetime import date, timedelta
+import screen_brightness_control as sbc
+import os
+import stat
 
 BASE_URL = 'https://api.olhar.media/'
-
+BASE_DIRECTRY = './assets/logs/'
 service = Services()
 
-logger = service.get_logger('main')
+current_city = service.get_current_city()
 
-def main():
-    # try:
-    #     logger.info('Fetching script...')
-    #     script = service.fetch_script(f'{BASE_URL}?getconfigupdate&equipid=1')
-    #     if script:
-    #         if service.is_valid_bash(script):
-    #             with open('script.sh', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_bash('./script.sh')
-    #             os.remove('./script.sh')
-    #             logger.info('Script executed successfully')
-    #         elif service.is_valid_python(script):
-    #             with open('script.py', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_python('./script.py')
-    #             os.remove('./script.py')
-    #             logger.info('Script executed successfully')
-    # except Exception as e:
-    #     logger.error(e)
 
-    try:   
-        logger.info('Fetching video data...')
-        return service.fetch_json(f'{BASE_URL}?getvideos&equipid=1')
+logging.basicConfig(filename=f'{BASE_DIRECTRY}{date.today()}.log', level=logging.DEBUG)
+logger = logging.getLogger('main')
+
+
+
+def delete_old_logs():
+    for filename in os.scandir(BASE_DIRECTRY):
+        if not filename.is_file():
+            continue
+        date_created = filename.name[:10]
+        week_ago = date.today() - timedelta(days=7)
+        if date_created == str(week_ago):
+            os.remove(os.path.join(BASE_DIRECTRY, filename.name))
+
+def filter_video_data(video_data):
+    new_video_data = []
+    for i in video_data:
+        if 'locations' not in i:
+            logger.error(f"Video object {i} has no 'locations' attribute.")
+            continue
+        for city in i['locations']:
+            if 'enname' not in city:
+                logger.error(f"City object {city} has no 'enname' attribute.")
+                continue
+            if city['enname'] == current_city:
+                new_video_data.append(i)
+    return new_video_data
+
+def run_maintenance_script():
+    try:
+        logger.info('Fetching script...')
+        script = service.fetch_script(f'{BASE_URL}?getconfigupdate&equipid=1')
+        if script:
+            with open('./maintenance.txt', 'w') as f:
+                f.write(script)
+            os.chmod('./maintenance.txt', stat.S_IRWXU)
+            os.system('bash ./maintenance.txt')
```

Comment:
This is how it should look like on Linux. Also, you never use `os.system`, you use the `subprocess` library instead.
You may want to change the `script_path` value, but you definitely should add the result to `.gitignore` if it ends up being a local path (which I so far expect).
```suggestion
            script_path = "./maintenance_script"
            with open(script_path, 'w') as f:
                f.write(script)
            os.chmod(script_path, stat.S_IRWXU)
            subprocess.run([script_path])
```

Reply:
✅